blogposts/2022/release-3.39

```diff
@@ -0,0 +1,65 @@
+---
+title: 'Sourcegraph 3.39 release'
+publishDate: 2022-04-21T10:00-07:00
+description: 'Sourcegraph 3.39 introduces Notebooks, Code Insights on native Postgres, faster Code Intelligence for Java, and dependencies search for Go.'
+tags: [blog, release]
+slug: 'release/3.39'
+published: true
+heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/3.38/sourcegraph-3-39-release.png
+socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/3.38/sourcegraph-3-39-release.png
+changelogItems:
+  - description: "A redesigned repository page, with information such as recent commits and Code Intelligence availability, is now available using the `new-repo-page` feature flag."
+    url: https://github.com/sourcegraph/sourcegraph/pull/33319
+    category: Repositories
+  - description: "The Code Insights GraphQL API now accepts search contexts as filters. Using a search context as a filter will extract the `repo` and `-repo` search query fields from the context and apply it, allowing you to scope your insights."
+    url: https://github.com/sourcegraph/sourcegraph/pull/33866
+    category: Code Insights
+  - description: "The Code Insights commit indexer can now index commits in smaller batches to ensure it succeeds for repositories with many commits. Set the number of days per batch in the site setting `insights.commit.indexer.windowDuration`. A value of 0 (default) will disable batching."
+    url: https://github.com/sourcegraph/sourcegraph/pull/33666
+    category: Code Insights
+  - description: "Code Insights series data is now sorted by semantic version and then alphabetically to make it easier to understand data in order of earliest to latest versions."
+    url: https://github.com/sourcegraph/sourcegraph/pull/33454
+    category: Code Insights
+  - description: "We've optimized search patterns containing `and` and `not` expressions. These kinds of queries are now an order of magnitude faster. Previous cases where no results were returned due to hitting the file limit should now work and return results quickly."
+    url: https://github.com/sourcegraph/sourcegraph/pull/33308
+    category: Search
+  - description: "Built-in authentication (i.e. username and password) now supports account lockout after consecutive failed sign-in attempts. New config options have been added under `\auth.lockout\` to customize the threshold for failed attempts and length of lockouts."
+    url: https://github.com/sourcegraph/sourcegraph/pull/33999
+    category: Admin
+---
+
+Sourcegraph 3.39 is now available! In this release:
+
+## Onboard developers fast with Notebooks, now Generally Available
+
+Built with onboarding in mind, Notebooks make documentation effortless, stable, and fun. Notebooks are inspired by the Jupyter Notebooks project and feature four unique block types, an intelligent notepad for creating notebooks on the fly, and deep integration with Sourcegraph's code search so you can always find the code you need.
```

Comment:
@benvenker — thoughts on this? It gives a little more technical context to what Notebooks are up front. I'm also trying to figure out the right way to say that Notebooks are "evergreen" (which I think is what you were getting at when you said "stable").

```suggestion
Built with onboarding in mind, Notebooks allow you to integrate search queries and Markdown to make documentation that is both effortless and doesn't get stale. Notebooks are inspired by the Jupyter Notebooks project and feature four unique block types, an intelligent notepad for creating notebooks on the fly, and deep integration with Code Search so you can always find the code you need.
```

Reply:
Maybe “search results” instead of “search queries” for the integration line? Small difference but feels better to me.

Reply:
```suggestion
Built with onboarding in mind, Notebooks allow you to integrate search results and Markdown to make documentation that is both effortless and doesn't get stale. Notebooks are inspired by the Jupyter Notebooks project and feature four unique block types, an intelligent notepad for creating notebooks on the fly, and deep integration with Code Search so you can always find the code you need.
```